structuredheadings/plugin.js

```diff
@@ -305,10 +307,12 @@
           setLevel(editor, heading);
           setCurrentStyle(editor, heading, value);
         },
+        onClick: function (value) {  // eslint-disable-line max-statements
+          editor.fire("saveSnapshot");
```

Comment:
I was unsure about this, but it looks like the built in style combo does this.  I assume it's for if you have some pending non-snapshotted changes before you click the dropdown.  It appears that if you truly do commit duplicate snapshots, they don't actually get created (or get merged somehow)